### PR TITLE
Cloud azure plugin

### PR DESCRIPTION
**Pull request by bogensberger**

### CODE REVIEW

**Comment by mikethebeer** [comment]
LGTM

File: plugins/cloud-azure/src/test/java/org/elasticsearch/discovery/azure/AzureSimpleTests.java [hunk 1]
```diff
@@ -20,29 +20,33 @@
...
     public AzureSimpleTests() {
         super(AzureComputeServiceSimpleMock.TestPlugin.class);
     }
 
+
     @Test
     public void one_node_should_run_using_private_ip() {
         Settings.Builder settings = Settings.settingsBuilder()
-                .put(Management.SERVICE_NAME, "dummy")
+        .put(Management.RESOURCE_GROUP_NAME, "crate-production")
```
Comment (mfussenegger) [inline plugins/cloud-azure/src/test/java/org/elasticsearch/discovery/azure/AzureSimpleTests.java]: probably should use a different name if this should get merged upstream

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java [hunk 2]
```diff
@@ -19,82 +19,91 @@
...
 import static org.elasticsearch.cloud.azure.management.AzureComputeService.Management.*;
 
-/**
- *
- */
 public class AzureComputeServiceImpl extends AbstractLifecycleComponent<AzureComputeServiceImpl>
-    implements AzureComputeService {
+        implements AzureComputeService {
+
+    public static class TestPlugin extends Plugin {
```
Comment (mfussenegger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: TestPlugin inside non-test code?

**Comment by mfussenegger** [comment]
Lgtm - but I just glanced over it

**Pull request merged**